news/3979.bug

```diff
@@ -0,0 +1,4 @@
+Do not bound the database session to the TransactionalSessionMaker class.
+Since Threads are sharing the memory binding the session object to the class
+make it possible for threads to override a previous session leading to unexpected
```

Comment:
"...to the class, it makes it possible for threads..."?

Reply:
+1 will update that

Reply:
If we go via the suggestion above, then we may want to drop the "to the class" in the line above this one